qutip/qip/device/processor.py

```diff
@@ -738,6 +739,11 @@ def run_state(self, init_state=None, analytical=False, states=None,
         else:
             kwargs["c_ops"] = sys_c_ops
 
+        # set the max step size as 1/2 of the smallest gate time.
+        options = kwargs.get("options", Options())
+        if options.max_step == 0.:
+            options.max_step = np.min(np.diff(self.get_full_tlist())) / 22
```

Comment:
Why 22? :)

Reply:
Ah, that was a mistake, it should just be `/2`. But still, this does not fully solve the problem above.